README.md
# nice_button

[![pub package](https://img.shields.io/pub/v/nice_button.svg)](https://pub.dev/packages/nice_button) 
[![Donate](https://img.shields.io/badge/Donate-PayPal-green.svg)](https://paypal.me/devEnthusiast?locale.x=en_US) 

A Flutter package for Android and IOS to design nice buttons with no hassle in your app. Enjoy !

## Installation

Add the following to pubspec.yaml
```yaml
dependencies:
  ...
  nice_button: ^0.1.7
```

## Usage Example

import nice_button.dart

```dart
import 'package:nice_button/nice_button.dart';
```

### For normal buttons

```dart
NiceButton(
    width: 255,
    elevation: 8.0,
    radius: 52.0,
    text: "Login",
    background: yourColor,
    onPressed: () {
        print("hello");
        },
    ),
```

### For mini buttons

Icon is required in this case

```dart
NiceButton(
    mini: true,
    icon: Icons.home,
    background: yourColor,
    onPressed: () {
        print("hello");
    },
    )
```


### For gradient buttons

```dart
var firstColor = Color(0xff5b86e5), secondColor = Color(0xff36d1dc);

NiceButton(
    radius: 40,
    padding: const EdgeInsets.all(15),
    text: "Register",
    icon: Icons.account_box,
    gradientColors: [secondColor, firstColor],
    onPressed: () {},
    )
```


![Package demo](screenshot.png) 

**Refer to [example folder](example/README.md) and the source code for more information.**


CHANGELOG.md
## [0.1.0] 

- Initial build for Flutter nice button library

## [0.1.1] 

- Format code and some minor changes

## [0.1.2]

 - Update the documentation and some minor changes

 ## [0.1.3]

 - Update package description to provide more info

 ## [0.1.4]

 - Remove 'new' reserved keyword (this is not necessary since Dart 2.0)
 - Dart Style Guide says "Don't initialize variables to null"
 - Upgrade SDK version to 2.2 (Now you can use spread operators)
 - Some basic refactoring to improve code reusability and make easy to read


## [0.1.5]
 - Add fontSize parameter to button
 

## [0.1.6]
 - Add iconColor parameter to button
 
## [0.1.7]
 - you can set custom width for mini buttons
 

test/nice_button_test.dart
import 'package:flutter_test/flutter_test.dart';

import 'package:nice_button/nice_button.dart';

void main() {
  test('adds one to input values', () {
    final calculator = Calculator();
    expect(calculator.addOne(2), 3);
    expect(calculator.addOne(-7), -6);
    expect(calculator.addOne(0), 1);
    expect(() => calculator.addOne(null), throwsNoSuchMethodError);
  });
}


lib/NiceButton.dart
import 'package:flutter/material.dart';

@immutable
class NiceButton extends StatelessWidget {
  /// This is a builder class for a nice button
  ///
  /// Icon can be used to define the button design
  /// User can use Flutter built-in Icons or font-awesome flutter's Icon  final bool mini;
  final IconData icon;

  /// specify the color of the icon
  final Color iconColor;

  /// radius can be used to specify the button border radius
  final double radius;

  /// List of gradient colors to define the gradients
  final List<Color> gradientColors;

  /// This is the button's text
  final String text;

  /// This is the color of the button's text
  final Color textColor;

  /// User can define the background color of the button
  final Color background;

  /// User can define the width of the button
  final double width;

  /// Here user can define what to do when the button is clicked or pressed
  final Function onPressed;

  /// This is the elevation of the button
  final double elevation;

  /// This is the padding of the button
  final EdgeInsets padding;

  /// `mini` tag is used to switch from a full-width button to a small button
  final bool mini;

  /// This is the font size of the text
  final double fontSize;

  const NiceButton(
      {Key key,
      this.mini = false,
      this.radius = 4.0,
      this.elevation = 1.8,
      this.textColor = Colors.white,
      this.iconColor = Colors.white,
      this.width,
      this.padding = const EdgeInsets.all(12.0),
      @required this.onPressed,
      @required this.text,
      @required this.background,
      this.gradientColors = const [],
      this.icon,
      this.fontSize = 23.0})
      : super(key: key);

  bool get existGradientColors => gradientColors.length > 0;

  LinearGradient get linearGradient => existGradientColors
      ? LinearGradient(
          colors: gradientColors,
          begin: Alignment.topLeft,
          end: Alignment.topRight)
      : LinearGradient(colors: [background, background]);

  BoxDecoration get boxDecoration => BoxDecoration(
      gradient: linearGradient,
      borderRadius: BorderRadius.circular(radius),
      color: background);

  TextStyle get textStyle => TextStyle(
      fontFamily: 'Montserrat',
      color: textColor,
      fontSize: fontSize,
      fontWeight: FontWeight.bold);

  Widget createContainer(BuildContext context) => mini
      ? Container(
          decoration: boxDecoration,
          width: width ?? 65.0,
          height: width ?? 65.0,
          child: Icon(icon, color: iconColor ?? Colors.white),
        )
      : Container(
          padding: padding,
          decoration: boxDecoration,
          constraints: BoxConstraints(
              maxWidth: width ?? MediaQuery.of(context).size.width / 1.5),
          child: Row(
            mainAxisAlignment: MainAxisAlignment.spaceEvenly,
            crossAxisAlignment: CrossAxisAlignment.center,
            children: <Widget>[
              Text(
                text,
                textAlign: TextAlign.center,
                style: textStyle,
              ),
              if (icon != null)
                Icon(
                  icon,
                  color: Colors.white,
                ),
            ],
          ),
        );

  @override
  Widget build(BuildContext context) {
    return FlatButton(
      materialTapTargetSize: MaterialTapTargetSize.shrinkWrap,
      shape:
          RoundedRectangleBorder(borderRadius: BorderRadius.circular(radius)),
      onPressed: onPressed,
      child: Material(
        color: Colors.transparent,
        type: MaterialType.card,
        borderRadius: BorderRadius.circular(radius),
        key: key,
        elevation: elevation,
        child: createContainer(context),
      ),
    );
  }
}


lib/NIceButtonBuilder.dart
import 'package:flutter/material.dart';

@immutable
class NiceButtonBuilder extends StatelessWidget {
  /// This is a builder class for a nice button
  ///
  /// Icon can be used to define the button design
  /// User can use Flutter built-in Icons or font-awesome flutter's Icon  final bool mini;
  final IconData icon;

  /// radius can be used to specify the button border radius
  final double radius;

  /// List of gradient colors to define the gradients
  final List<Color> gradientColors;

  /// This is the button's text
  final String text;

  /// This is the color of the button's text
  final Color textColor;

  /// User can define the background color of the button
  final Color background;

  /// User can define the width of the button
  final double width;

  /// Here user can define what to do when the button is clicked or pressed
  final Function onPressed;

  /// This is the elevation of the button
  final double elevation;

  /// This is the padding of the button
  final EdgeInsets padding;

  /// `mini` tag is used to switch from a full-width button to small button
  final bool mini;

  const NiceButtonBuilder(
      {Key key,
      this.mini = false,
      this.radius = 4.0,
      this.elevation = 3.0,
      this.textColor = Colors.white,
      this.width = null,
      this.padding = const EdgeInsets.all(15.0),
      @required this.onPressed,
      @required this.text,
      @required this.background,
      this.gradientColors = const [],
      this.icon})
      : super(key: key);

  @override
  Widget build(BuildContext context) {
    return Material(
      key: key,
      elevation: elevation,
      borderRadius: BorderRadius.circular(radius),
      child: InkWell(
          onTap: onPressed,
          child: Container(
              width: MediaQuery.of(context).size.width,
              decoration: BoxDecoration(
                  color: background,
                  borderRadius: BorderRadius.circular(40.0),
                  gradient: gradientColors.length > 0
                      ? LinearGradient(
                          colors: gradientColors,
                          begin: Alignment.topLeft,
                          end: Alignment.topRight)
                      : LinearGradient(colors: [background])),
              padding: padding,
              child: mini
                  ? Container(
                      width: 35.0,
                      height: 35.0,
                      child: Icon(
                        icon,
                        color: Colors.white,
                      ),
                    )
                  : Container(
                      constraints: BoxConstraints(
                        maxWidth:
                            width ?? MediaQuery.of(context).size.width / 1.5,
                      ),
                      child: Center(
                        child: Row(
                          children: <Widget>[
                            Padding(
                              padding: EdgeInsets.symmetric(
                                  horizontal:
                                      MediaQuery.of(context).size.width / 20),
                              child: Icon(
                                icon,
                                color: Colors.white,
                              ),
                            ),
                            Text(
                              text,
                              textAlign: TextAlign.center,
                              style: TextStyle(
                                  fontFamily: 'Montserrat',
                                  color: textColor,
                                  fontSize: 25.0,
                                  fontWeight: FontWeight.bold),
                            ),
                          ],
                        ),
                      ),
                    ))),
    );
  }
}


example/lib/main.dart
import 'package:flutter/material.dart';
import 'package:nice_button/NiceButton.dart';

void main() => runApp(MyApp());

class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Nice Button Demo',
      theme: ThemeData(
        primarySwatch: Colors.blue,
      ),
      home: MyHomePage(),
      debugShowCheckedModeBanner: false,
    );
  }
}

class MyHomePage extends StatefulWidget {
  @override
  MyHomePageState createState() {
    return MyHomePageState();
  }
}

class MyHomePageState extends State<MyHomePage> with TickerProviderStateMixin {
  @override
  Widget build(BuildContext context) {
    var firstColor = Color(0xff5b86e5), secondColor = Color(0xff36d1dc);
    return Scaffold(
      appBar: AppBar(
        backgroundColor: firstColor,
        title: Text('Nice Button Demo'),
      ),
      body: Material(
        child: Center(
          child: Padding(
            padding: const EdgeInsets.all(8.0),
            child: Column(
              mainAxisAlignment: MainAxisAlignment.spaceEvenly,
              children: <Widget>[
                NiceButton(
                  // width: 515,
                  elevation: 8.0,
                  radius: 52.0,
                  text: "Login",
                  background: firstColor,
                  onPressed: () {
                    print("hello");
                  },
                ),
                NiceButton(
                  mini: true,
                  icon: Icons.home,
                  background: firstColor,
                  onPressed: () {
                    print("hello");
                  },
                ),
                NiceButton(
                  elevation: 5,
                  mini: true,
                  icon: Icons.label,
                  gradientColors: [secondColor, firstColor],
                  onPressed: () {},
                ),
                NiceButton(
                  radius: 40,
                  padding: const EdgeInsets.all(15),
                  text: "Register",
                  icon: Icons.account_box,
                  gradientColors: [secondColor, firstColor],
                  onPressed: () {},
                )
              ],
            ),
          ),
        ),
      ),
    );
  }
}
